tests/integration/devfile/cmd_devfile_url_test.go

```diff
@@ -70,19 +70,8 @@ var _ = Describe("odo devfile url command tests", func() {
 			Expect(stdout).To(ContainSubstring("host must be provided"))
 
 			helper.CmdShouldPass("odo", "url", "create", url1, "--port", "3000", "--host", host, "--ingress")
-			helper.CmdShouldPass("odo", "push")
-			helper.WaitForCmdOut("odo", []string{"url", "list"}, 1, false, func(output string) bool {
-				if strings.Contains(output, url1) {
-					Expect(output).Should(ContainSubstring(url1 + "." + host))
-					return true
-				}
-				return false
-			})
-			helper.CmdShouldPass("odo", "url", "delete", url1, "-f")
```

Comment:
@yangcao77 Does ```odo url delete url1 -f``` is supported for URLs defined in devfile ? if yes you can add these validation at the end of the spec.

Reply:
@amitkrout It works for manually created URLs, but not for automatically created URLs from endpoint definition in devfile. 
`odo url delete -f` is tested many times in this file. I would say that it is ok to skip it in this test. The point of this test is to make sure that URL is listed  in the `odo push` output

Reply:
@kadel 
> It works for manually created URLs, but not for automatically created URLs from endpoint definition in devfile.

Then i assume that it is same issue like url listing which does not work for URLs defined in devfile. 

> odo url delete -f is tested many times in this file. I would say that it is ok to skip it in this test. The point of this test is to make sure that URL is listed in the odo push output

My concern is why the existing steps is removed without any reason like #3615, may be the steps has been added initially for certain reason. Its ok for me if the same functionalities is verified in in other spec 👍

Reply:
Yes. The same reason like url listing and describe for URLs defined in devfile. 
It is expected to resolve in issue https://github.com/openshift/odo/issues/3718
Since we want to move the url creation and deletion into devfile.yaml.